content/en/blog/_posts/2023-04-11-kubernetes-1.27-blog.md

```diff
@@ -0,0 +1,223 @@
+
+
+## Respect PodTopologySpread after rolling upgrades
+
+`matchLabelKeys` is a list of pod label keys to select the pods over which spreading will be calculated. The keys are used to lookup values from the pod labels, those key-value labels are ANDed with `labelSelector` to select the group of existing pods over which spreading will be calculated for the incoming pod. Keys that don't exist in the pod labels will be ignored. A null or empty list means only match against the `labelSelector`.
+
+With `matchLabelKeys`, users don't need to update the `pod.spec` between different revisions. The controller/operator just needs to set different values to the same `label` key for different revisions. The scheduler will assume the values automatically based on `matchLabelKeys`. For example, if users use Deployment, they can use the label keyed with `pod-template-hash`, which is added automatically by the Deployment controller, to distinguish between different revisions in a single Deployment.
+
+
+## Faster SELinux volume relabeling using mounts
+
+In this release, how SELinux labels are applied to volumes used by Pods is graduating to beta. This feature speeds up container startup by mounting volumes with the correct SELinux label instead of changing each file on the volumes recursively. Linux kernel with SELinux support allows the first mount of a volume to set SELinux label on the whole volume using `-o context=` mount option. This way, all files will have assigned the given label in a constant time, without recursively walking through the whole volumes.
+
+The `context` mount option cannot be applied to bind mounts or re-mounts of already mounted volumes.
+For CSI storage, a CSI driver does the first mount of a volume, and so it must be the CSI driver that actually
+applies this mount option. We added a new field `SELinuxMount` to CSIDriver objects, so that drivers can
+announce whether they support the `-o context` mount option.
+
+If Kubernetes knows the SELinux label of a Pod **and** the  CSI driver responsible for a pod's volume
+announces `SELinuxMount: true` **and** the volume has Access Mode `ReadWriteOncePod`, then it
+will ask the CSI driver to mount the volume with mount option `context=` **and** it will tell the container
+runtime not to relabel content of the volume (because all files already have the right label).
+Get more information on this from the KEP: [Speed up SELinux volume relabeling using mounts](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1710-selinux-relabeling).
+
+## Robust VolumeManager reconstruction goes to beta
+
+This is a volume manager refactoring that allows the kubelet to populate additional information about how
+existing volumes are mounted during the kubelet startup. In general, this makes volume cleanup more robust. 
+By adding `NewVolumeManagerReconstruction` feature gate and enabling it by default will enhance the discovery of mounted volumes during kubelet startup.
```

Comment:
```suggestion
Adding `NewVolumeManagerReconstruction` feature gate and enabling it by default will enhance the discovery of mounted volumes during kubelet startup.
```

Reply:
The revised doesn't make sense: the announcement goes out _after_ contributors have added that feature gate into the Kubernetes code.
How about:
```suggestion
If you enable the `NewVolumeManagerReconstruction` feature gate on a node, you'll get enhanced discovery of
mounted volumes during kubelet startup.
```